typing/parmatch.ml

```diff
@@ -793,7 +794,7 @@ let build_specialized_submatrices ~extend_row discr rows =
     let initial_constr_group =
       let open Patterns.Head in
       match discr.pat_desc with
-      | Record _ | Tuple _ | Lazy ->
```

Comment:
Should unboxed tuple also be added to this case? (I have not thought deeply about what this case is doing, and the comment suggests it's a non-critical improvement in any case, so don't spend too much time trying to answer this)

Reply:
I'm also not sure exactly what this case is doing, but I just added it because unboxed tuples/records should be treated analogously to their boxed versions in this file.